### PR TITLE
Fix dialog overlay potentially pushing dialog while not loaded

### PR DESCRIPTION
**Pull request by frenzibyte**
As seen in https://github.com/ppy/osu/runs/7324796490?check_suite_focus=true.

The cause of this is `DialogOverlay.Push()` adding the dialog to `dialogContainer` while that container is in `Loading` state. This can rarely occur, but can be simulated by adding a wait step before the dialog push step and make `dialogContainer` stuck on `Loading` state by sleeping on its BDL.

Not sure why the push is not always scheduled, but went with the simplest fix rather than changing it and end up with more issues. Open for discussion.

### CODE REVIEW

**Comment by peppy** [comment]
Did you try just making it always schedule? Is there an issue with doing so? If not I would prefer that for simplicity.

**Comment by frenzibyte** [comment]
Thought there was a specific reason to it not always scheduling. Will give that a try.

**Comment by peppy** [comment]
I've applied a bit more restructuring as I still found the hide event / flow logic to be very hard to follow. Can you check whether this reads better to you now?

**Review by peppy — DISMISSED** [review]

**Comment by frenzibyte** [comment]
Reads much better 👍

**Review by peppy — APPROVED** [review]
